task/buildah-remote/0.1/buildah-remote.yaml

```diff
@@ -0,0 +1,396 @@
+        touch /var/lib/containers/java
+      fi
+
+      # Fixing group permission on /var/lib/containers
+      chown root:root /var/lib/containers
+
+      sed -i 's/^\s*short-name-mode\s*=\s*.*/short-name-mode = "disabled"/' /etc/containers/registries.conf
+
+      # Setting new namespace to run buildah - 2^32-2
+      echo 'root:1:4294967294' | tee -a /etc/subuid >> /etc/subgid
+
+      if [ "${HERMETIC}" == "true" ]; then
+        BUILDAH_ARGS="--pull=never"
+        UNSHARE_ARGS="--net"
+        for image in $(grep -i '^\s*FROM' "$dockerfile_path" | sed 's/--platform=\S*//' | awk '{print $2}'); do
+          unshare -Ufp --keep-caps -r --map-users 1,1,65536 --map-groups 1,1,65536 -- buildah pull $image
+        done
+        echo "Build will be executed with network isolation"
+      fi
+
+      if [ -n "${PREFETCH_INPUT}" ]; then
+        mv cachi2 /tmp/
+        chmod -R go+rwX /tmp/cachi2
+        VOLUME_MOUNTS="--volume /tmp/cachi2:/cachi2"
+        sed -i 's|^\s*run |RUN . /cachi2/cachi2.env \&\& \\\n    |i' "$dockerfile_path"
+        echo "Prefetched content will be made available"
+      fi
+
+      LABELS=(
+        "--label" "build-date=$(date -u +'%Y-%m-%dT%H:%M:%S')"
+        "--label" "architecture=$(uname -m)"
+        "--label" "vcs-type=git"
+      )
+      [ -n "$COMMIT_SHA" ] && LABELS+=("--label" "vcs-ref=$COMMIT_SHA")
+      [ -n "$IMAGE_EXPIRES_AFTER" ] && LABELS+=("--label" "quay.expires-after=$IMAGE_EXPIRES_AFTER")
+
+      unshare -Uf $UNSHARE_ARGS --keep-caps -r --map-users 1,1,65536 --map-groups 1,1,65536 -- buildah build \
+        $VOLUME_MOUNTS \
+        $BUILDAH_ARGS \
+        ${LABELS[@]} \
+        --tls-verify=$TLSVERIFY --no-cache \
+        --ulimit nofile=4096:4096 \
+        -f "$dockerfile_path" -t $IMAGE $CONTEXT
+
+      container=$(buildah from --pull-never $IMAGE)
+      buildah mount $container | tee /workspace/container_path
+      echo $container > /workspace/container_name
+
+      # Save the SBOM produced by Cachi2 so it can be merged into the final SBOM later
+      if [ -n "${PREFETCH_INPUT}" ]; then
+        cp /tmp/cachi2/output/bom.json ./sbom-cachi2.json
+      fi
+
+      buildah push $IMAGE oci:rhtap-final-image
+      REMOTESSHEOF
+      chmod +x scripts/script-build.sh
+      rsync -ra scripts $SSH_HOST:$BUILD_DIR
+      ssh $SSH_ARGS $SSH_HOST $PORT_FORWARD podman  run $PODMAN_PORT_FORWARD -e BUILDAH_FORMAT=oci -e STORAGE_DRIVER=vfs -e HERMETIC=$(params.HERMETIC) -e PREFETCH_INPUT=$(params.PREFETCH_INPUT) -e CONTEXT=$(params.CONTEXT) -e DOCKERFILE=$(params.DOCKERFILE) -e IMAGE=$(params.IMAGE) -e TLSVERIFY=$(params.TLSVERIFY) -e IMAGE_EXPIRES_AFTER=$(params.IMAGE_EXPIRES_AFTER) -e COMMIT_SHA=$(params.COMMIT_SHA)  --rm  -v $BUILD_DIR/workspaces/source:$(workspaces.source.path):Z  -v $BUILD_DIR/scripts:/script:Z --user=0  $(params.BUILDER_IMAGE)  /script/script-build.sh
```

Comment:
Unquoted variables can break things, allow shell injection etc. Especially Tekton `$(params.*)` - it's not even possible to quote them correctly. If nothing else, PREFETCH_INPUT may contain whitespace.

Params should be [injected via env](https://github.com/redhat-appstudio/build-definitions/blob/082637759f23a43d694823aa5a7fc1316c0f7c01/task/buildah/0.1/buildah.yaml#L70-L88) and referenced as properly quoted env vars